### ISSUE TITLE
Tensorboard doesn't work

### ISSUE THREAD

**Issue by KKamojjala** [rstudio/tensorflow#420]
Hi,

I get the following error when I try to run the tensorboard command

**Comment by KKamojjala** [comment]
I found a workaround for this issue. I am not sure anyone else has this issue, but I would like to post the not-so-efficient way to visualize your tensorbard.

Here is how typically the log files are stored to be visualized using tensorflow

history <- model %>% fit(
  normalized_train_data, train_label, 
  epochs = 30, batch_size = 128,   
  verbose = 1,
  callbacks = callback_tensorboard("logs/run_a"),
  validation_data = list(normalized_validation_data,validation_label)
)

But, when I run the tensorboard(log_dir = "logs/run_a") command in rstudio, I get the following error

Error: invalid version specification ‘2020-05-28 22:40:18.752287: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_101.dll’

So, here is what I did.
From Rstudio --> Tools --> Shell

This pops up a command prompt, and inside it, just type "tensorboard --logdir logs/run_a"

![Image1](https://user-images.githubusercontent.com/6326195/83219062-fc362680-a134-11ea-917e-18cc479d2a90.PNG)

This will say "Serving TensorBoard on localhost:..
Tensorboard 2.2.1 at http://localhost:6006/

Just copy this link "http://localhost:6006/" into your browser and you should be able to access TensorBoard there.

I know this is not the way but atleast I am not stuck at this point. Please let me know if anyone else finds the correct solution.

**Comment by faltinl** [comment]
Hi,-
thanks for your comment which looks very useful indeed. Accessing my browser through tensorboard would be exactly what I was hoping for!-)

However, if I do as indicated (after starting with library(keras) in RStudio, as indicated in https://github.com/rstudio/tensorflow/issues/407) I get the message "The command "tensorboard" is either misspelled or could not be found." 

It might be a question into which directory the command has been written. My default in this case, appearing when opening the shell prompt, is the current RStudio working directory. How's that in your case?

**Comment by KKamojjala** [comment]
The shell prompt in my case also opens the current RStudio working directory and the command tensorboard just runs fine.

Does tensorboard command run in your RStudio console ? when you open the shell prompt, what does it tell you when you run "pip show tensorboard" ?

**Comment by faltinl** [comment]
My wd is the directory where my source is located. Quite logically, for `pip show tensorboard` it indicates "the command pip is either misspelled or does not exist", because pip is a python command and that can only work correctly if the wd would be the virtual python environment.
If I look there (this, however, requires to change to the 'Anaconda Power Shell Prompt (Miniconda3)'), then the command works properly and I get the answer 
```
Name: tensorboard
Version: 2.1.1.
etc. (various info regarding tensorboard)
```
To me it looks as if I could not change the working directory provided by the RStudio Shell to this virtual environment.

Which directory is your working directory?

**Comment by KKamojjala** [comment]
My working directory is same as what you do. I also set it manually in every code that I run using
setwd("Folder I am working in") 

Its weird.  I use windows 10. What do you use? Sorry, I know this error is annoying but now, I just run tensorboard from the command prompt rather than inside RStudio where it still throws me an error.

You said, tensorboard runs fine from the anaconda shell prompt. I tried the same thing. I opened the  'Anaconda Power Shell Prompt (Miniconda3)' and ran the tensorboard command by providing the whole path for the log directory as shown below and it still runs fine

![Image2](https://user-images.githubusercontent.com/6326195/83319922-083ee880-a208-11ea-9a4e-03332cc5122b.PNG)

Let me know if you want me to try something else and I would be happy to help !

**Comment by faltinl** [comment]
Hi - and thank you so much for your interest! Yes, I'm using WIN10, too.

However, neither from RStudio Shell (even when set to Option 'Windows Power Shell') nor from RStudio Terminal it is possible for my to induce tensorboard to any action at all. So, whatever I'm typing in w.r.t. Python, R and in particular tensorboard, it doesn't recognize any commands. It is not possible to activate my virtual environment r-reticulate from there either. This may have to do with the fact that I'm using Python exclusively through virtual environments. 

Using the 'Anaconda Power Shell Prompt (Miniconda3)' outside RStudio, I have first to 'conda activate r-reticulate', i.e. my virtual environment, otherwise nothing would happen either. In activated mode, it is possible to use pip..., in particular pip list, pip install... and pip show...

From that I learned that I am using tensorboard v2.1.1. - but I didn't succeed to upgrade to your v2.2.1. This is probably due to the fact that the present tensorboard installation was done implicitely by way of Keras for R (it doesn't appear in the RStudio package list, for example) and I am rather reticent to intrude into this sphere - it has cost me already too much time to get Keras for R and my GPU working at all...

What did work was 'tensorboard --helpfull'. However, 'tensorboard --logdir mydir' first looked good, the dynamic lib was opened successfully, but instead of proceeding, as in your case, to show a local host address, an error message with an endless traceback appeared. The key error was indicated as some problem with package `protobuf`, which I confirmed to be correctly installed and 'fullfiling al requirements'. 

So, that's my present state of the tensorboard art and I think for the moment I'll have to leave it there. I really appreciate you support and willingness to help - thanks a lot!

**Comment by venciso** [comment]
Thanks for posting this @KKamojjala! I had the same problem and I can use your workaround.
